src/alu.ts

```diff
@@ -1524,8 +1524,10 @@ export class Reduction implements FpHashable {
   evaluate(...values: any) {
     if (this.dtype === DType.Bool) {
       if (this.op === AluOp.Add || this.op === AluOp.Max) {
-        return values.reduce((a: boolean, b: boolean) => a || b, true);
+        // OR reduction: identity is false
+        return values.reduce((a: boolean, b: boolean) => a || b, false);
```

Comment:
Thanks, this bug is indeed an issue (although not exercised in the current code as you pointed out)